testing/sample_framework/src/ios/ios_app_framework.mm

```diff
@@ -47,10 +47,19 @@ @interface FTAViewController : UIViewController
 // Test Loop on iOS doesn't provide the app under test a path to save logs to, so set it here.
 #define GAMELOOP_DEFAULT_LOG_FILE "Results1.json"
 
-static int g_exit_status = 0;
-static bool g_shutdown = false;
-static NSCondition *g_shutdown_complete;
-static NSCondition *g_shutdown_signal;
+enum class RunningStatus {
+  kRunning,
+  kShuttingDown,
+  kShutDown
+};
+
+// Note: g_running_status and g_exit_status must only be accessed while holding the lock from
+// g_running_status_condition; also, any changes to these values should be followed up with a
+// call to [g_running_status_condition broadcast].
+static NSCondition *g_running_status_condition;
+static RunningStatus g_running_status = RunningStatus::kRunning;
+static int g_exit_status = -1;
```

Comment:
Nit: it seems a little inconsistent that `status` is used in two closely related variables but has slightly different meanings. How about `s/RunningStatus/RunningState/`?